src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -710,11 +710,19 @@ private function resolveServices($value, $file, $isParameter = false)
                 }
             }
             if ('tagged' === $value->getTag()) {
-                if (!\is_string($argument) || !$argument) {
-                    throw new InvalidArgumentException(sprintf('"!tagged" tag only accepts non empty string in "%s".', $file));
+                if (\is_string($argument) && $argument) {
+                    return new TaggedIteratorArgument($argument);
                 }
 
-                return new TaggedIteratorArgument($argument);
+                if (\is_array($argument) && isset($argument['tag']) && $argument['tag']) {
+                    if ($diff = array_diff(array_keys($argument), ['tag', 'index_by', 'default_index_method'])) {
+                        throw new InvalidArgumentException(sprintf('"!tagged" tag contains unsupported key "%s"; supported ones are "tag", "index_by" and "default_index_method".', implode(', ', $diff)));
```

Comment:
`implode('", "', $diff)`?

Reply:
to list all keys that are not supported by the !tagged.

Reply:
Yes, what I mean is also add quotes next to the comma so that each key will be enclosed by its own set of quotes.

Reply:
yes , indeed a mistake, thank you